airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_persons.py

```diff
@@ -0,0 +1,632 @@
+        case "events":
+            return StripeRequestBuilder.events_endpoint(_ACCOUNT_ID, _CLIENT_SECRET)
+        case "persons":
+            return StripeRequestBuilder.persons_endpoint(_ACCOUNT_ID, _CLIENT_SECRET)
+        case _:
+            raise ValueError("Invalid resource specified")
+
+
+def _create_response() -> HttpResponseBuilder:
+    return create_response_builder(
+        response_template=find_template("accounts", __file__),
+        records_path=FieldPath("data"),
+        pagination_strategy=StripePaginationStrategy()
+    )
+
+
+def _create_record(resource: str) -> RecordBuilder:
+    return create_record_builder(
+        find_template(resource, __file__),
+        FieldPath("data"),
+        record_id_path=FieldPath("id"),
+        record_cursor_path=FieldPath("created")
+    )
+
+
+def emits_successful_sync_status_messages(status_messages: List[AirbyteStreamStatus]) -> bool:
+    return (len(status_messages) == 3 and status_messages[0] == AirbyteStreamStatus.STARTED
+            and status_messages[1] == AirbyteStreamStatus.RUNNING and status_messages[2] == AirbyteStreamStatus.COMPLETE)
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class PersonsTest(TestCase):
+    @HttpMocker()
+    def test_full_refresh(self, http_mocker):
+        http_mocker.get(
+            _create_request("accounts").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("accounts")).build(),
+        )
+
+        http_mocker.get(
+            _create_request("persons").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("persons")).with_record(record=_create_record("persons")).build(),
+        )
+
+        http_mocker.get(
+            _create_request("events").with_created_gte(_NOW - timedelta(days=30)).with_created_lte(_NOW).with_limit(100).with_types(["person.created", "person.updated", "person.deleted"]).build(),
+            _create_response().with_record(record=_create_record("events")).with_record(record=_create_record("events")).build(),
+        )
+
+        source = SourceStripe(config=_CONFIG, catalog=_create_catalog())
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert emits_successful_sync_status_messages(actual_messages.get_stream_statuses(_STREAM_NAME))
+        assert len(actual_messages.records) == 2
+
+    @HttpMocker()
+    def test_parent_pagination(self, http_mocker):
+        # First parent stream accounts first page request
+        http_mocker.get(
+            _create_request("accounts").with_limit(100).build(),
+            _create_response().with_record(record=_create_record("accounts")).with_pagination().build(),
+        )
+
+        # Second parent stream accounts second page request
+        http_mocker.get(
+            _create_request("accounts").with_limit(100).with_starting_after("acct_1G9HZLIEn49ers").build(),
```

Comment:
nit: I have a preference for configuring this in the first parent stream responses for a couple of reasons:
* Test is more expressive as you can see the flow of id `acct_1G9HZLIEn49ers`. Here, we can only assume it is configured somewhere in the first parent stream response
* It is more rebust. If someone was to change the response in the template, this test might fail while it's unrelated to the devs intent when he changed the template

Reply:
I'm not quite sure I follow what you're recommending. How would I be able to configure this in the first parent stream request/response mock? This second page request needs to cover the incoming request where `starting_after=acct_1G9HZLIEn49ers` is a query parameter and as far as i know the mocker doesn't support multiple requests, just multiple responses.

Unless I'm not understanding what you're recommending in which could you point me towards an example of what you meant? While implementing, I looked at a few other tests like `test_early_fraud_warnings.test_given_many_pages_when_read_then_return_records()` which followed a similar two mock pattern to test pagination

Reply:
Here is an example of how it is configured for early_fraud_warnings: https://github.com/airbytehq/airbyte/pull/33306/files#diff-00bd50afcc672f2737548c1ff387d3a09094e02e611e0dde8138898acdbab374R139-R146

Basically, the first page is configured to return the last record `.with_id("last_record_id_from_first_page")` and we can see that the second request is also performed with `.with_starting_after("last_record_id_from_first_page")`.

Reply:
ah okay i understand what you mean now. In that you want the test to define the ID which we'll use to paginate rather than rely on what is defined in the json. I'll update the pagination tests with that suggestion